### DISCUSSION TITLE
Vercel does not display not-found page on Next.js project

### DISCUSSION THREAD

**Discussion by dkottas-nelios** [vercel/community#7171] [Bug Report]
### Summary

I have a Next.js project that uses generateStaticParams with dynamicParams = false. Urls that don't use the provided params and should return thwe not found page, do so on localhost and Netlify but not on Vercel.
for example:
https://next-notfound-test.netlify.app/page-x/page-d
https://nextjs-notfound.vercel.app/page-x/page-d
Both pages are deployed from the same commit.

### Example

https://nextjs-notfound.vercel.app/this-does-not-exist/page-d

### Steps to Reproduce

Use this repo https://gitlab.com/dkottas/nextjs-notfound and deploy it on Vercel with one env variable NEXT_PUBLIC_DOMAIN="http://localhost:3000"
pointing to the address where you deploy.
You should note what is described above.

**Comment by dkottas-nelios** [comment]
The structure of the urls in the example is:
/[page]/[subpage]
When opening an existing subpage as child of a non-existing page, like: /wrong/right results in Netlify are not consostent while in Vercel the page always opens. When you open a non-existing page:
https://next-notfound-test.netlify.app/page-x
https://nextjs-notfound.vercel.app/page-x
Netlify shows the 404 page but Vercel does not and it does not add the noindex meta tag.
This is a serious issue on a production site since any random url may show empty but indexable pages. We need to know if there is a fix or else we will need to move to some other hosting service.

**Comment by pawlean** [comment]
Hi @dkottas-nelios!

Let me try and help you out here 😄 

My understanding is that your Next.js project using `generateStaticParams` with `dynamicParams = false` behaves inconsistently between local, Netlify, and Vercel deployments. Specifically, URLs that don't match the provided parameters correctly return the "not found" page on localhost and Netlify but not on Vercel.

However, when I went on both these deployments, they each return "not found". 

Netlify: [netlify-url](https://next-notfound-test.netlify.app/page-x/page-d)
Vercel: [vercel-url](https://nextjs-notfound.vercel.app/page-x/page-d)

Am I missing something? Below is a screenshot from your Vercel link:

![CleanShot 2024-07-08 at 16 36 57@2x](https://github.com/vercel/community/assets/17087373/f8a3ed9f-4a37-4932-88e2-a3953fdef460)

**Reply by dkottas-nelios** [reply]
I have been trying different solutions. Now I have used a middleware as in multilingual app.
https://nextjs-notfound.vercel.app/en
https://main--next-notfound-test.netlify.app/en
Page A is the one that should be working and Page X should display the not-found page.
Netlify works like localhost. Vercel shows a 500 error on not-found.

**Reply by pawlean** [reply]
I just tried both, and this is what I see. 

![CleanShot 2024-07-09 at 13 27 30](https://github.com/vercel/community/assets/17087373/2c3625a8-5cb5-4e95-86e6-da83a2844d6f)

**Reply by dkottas-nelios** [reply]
The homepage is fine. It is the Page X that is supposed to show the bot-found page:
https://nextjs-notfound.vercel.app/en/pages/page-x
https://main--next-notfound-test.netlify.app/en/pages/page-x

**Reply by pawlean** [reply]
Ah, yes! I see this now.

And you have the same 404 page created for both?

**Reply by dkottas-nelios** [reply]
It is the same repo, same branch.

**Reply by pawlean** [reply]
Could you share the repo with us? 🙏🏼

**Reply by dkottas-nelios** [reply]
Here: https://github.com/Cuboctaedro/nextjs-notfound

**Reply by pawlean** [reply]
I can confirm that I see this and reproduce. Looks like a bug, can you raise a report [here](https://github.com/vercel/next.js/issues/new?assignees=&labels=bug&projects=&template=1.bug_report.yml)?

[a comment by an automated account is not shown]
